addon/ng2/tasks/lib-install.js

```diff
@@ -18,50 +18,49 @@ module.exports = Task.extend({
   completionOKMessage: '',
   completionErrorMessage: 'Error installing packages. Did you misspelt it?',
 
-  init: function() {
-    this.npm = this.npm || require('npm');
-  },
   run: function(options) {
     this.packages = options.packages || [];
     this.skipInjection = options.skipInjection || false;
     this.autoInjection = options.autoInjection || false;
     this.disableLogger();
 
-    this.ui.startProgress(chalk.green('Installing 3rd party package:', 
+    this.ui.startProgress(chalk.green('Installing 3rd party package:',
       this.packages.join(', ')), chalk.green('.'));
 
-    this.npmOptions = {
-      logLevel: 'error',
-      logstream: this.ui.outputStream,
-      color: 'always',
-      optional: true,
-      'save-dev': true,
-      'save-exact': !!options['save-exact']
-    };
-
-    return npm('install', this.packages, this.npmOptions, this.npm)
-      .then(function(npmresp) {
-        if (this.skipInjection) {
-          return this.announceOKCompletion();
-        }
+    this.npmOptions = ' --loglevel error --color always --optional --save-dev ' +
+                      '--save-exact ' + !!options['save-exact'];
 
-        if (this.autoInjection) {
-          var pkg = this.packages[0];
-          
-          if (existsSync(path.resolve(process.cwd(), 'src', 'app.ts'))) {
-            var entryPoint = path.resolve(process.cwd(), 'src', 'app.ts');
-            var packageData = this.parseFile(pkg);
-            this.importInBootstrap(entryPoint, pkg, packageData);
-          }
+    var npmCommand = 'npm install ' + this.packages.join(' ') + this.npmOptions;
+
+    return new Promise(function (resolve, reject){
```

Comment:
This is a pattern that we will do a lot and I'd like to avoid the copy and paste. Could you move the sh.exec and promise into a helper function that would be imported everywhere?